docs/tutorial-code/redis.go

```diff
@@ -25,124 +25,82 @@ const (
-	if splittedLine[0] == "dbfilename" {
-		err = e.SetInventoryItem(splittedLine[0], "value", splittedLine[1])
-		if err != nil {
-			return err
-		}
-	}
-
-	cmd = exec.Command("/bin/sh", "-c", "redis-cli CONFIG GET bind")
-	output, err = cmd.CombinedOutput()
-	if err != nil {
-		return err
-	}
-	splittedLine = strings.Split(string(output), "\n")
-	if splittedLine[0] == "bind" {
-		err = e.SetInventoryItem(splittedLine[0], "value", splittedLine[1])
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-func populateMetrics(ms *metric.Set) error {
-	cmd := exec.Command("/bin/sh", "-c", "redis-cli info | grep instantaneous_ops_per_sec:")
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		return err
+		return 0, err
 	}
 	splittedLine := strings.Split(string(output), ":")
 	if len(splittedLine) != 2 {
-		return fmt.Errorf("cannot split the output line")
+		return 0, fmt.Errorf("Cannot split the output line")
 	}
-	metricValue, err := strconv.ParseFloat(strings.TrimSpace(splittedLine[1]), 64)
-	if err != nil {
-		return err
-	}
-	ms.SetMetric("query.instantaneousOpsPerSecond", metricValue, metric.GAUGE) // nolint: errcheck
-
-	cmd = exec.Command("/bin/sh", "-c", "redis-cli info | grep total_connections_received:")
-	output, err = cmd.CombinedOutput()
-	if err != nil {
-		return err
-	}
-	splittedLine = strings.Split(string(output), ":")
-	if len(splittedLine) != 2 {
-		return fmt.Errorf("cannot split the output line")
-	}
-	metricValue, err = strconv.ParseFloat(strings.TrimSpace(splittedLine[1]), 64)
-	if err != nil {
-		return err
-	}
-
-	ms.SetMetric("net.connectionsReceivedPerSecond", metricValue, metric.RATE) // nolint: errcheck
-	return nil
+	return strconv.ParseFloat(strings.TrimSpace(splittedLine[1]), 64)
 }
 
-func populateEvents(e *integration.Entity) error {
-	cmd := exec.Command("/bin/sh", "-c", "redis-cli info | grep uptime_in_seconds:")
+func queryRedisConfig(query string) (string, string) {
+	cmd := exec.Command("/bin/sh", "-c", fmt.Sprintf("redis-cli CONFIG GET %s", query))
+
 	output, err := cmd.CombinedOutput()
-	if err != nil {
-		return err
-	}
-	splittedLine := strings.Split(string(output), ":")
-	if len(splittedLine) != 2 {
-		return fmt.Errorf("cannot split the output line")
-	}
-	uptime, err := strconv.ParseFloat(strings.TrimSpace(splittedLine[1]), 64)
-	if err != nil {
-		return err
-	}
-	if uptime < 60 {
-		err = e.AddEvent(event.NewNotification("Redis Server recently started"))
-		if err != nil {
-			return err
-		}
-	}
-	if uptime < 60 {
-		err = e.AddEvent(event.New("Redis Server recently started", "redis-server"))
-	}
+	panicOnErr(err)
 
-	return err
+	splittedLine := strings.Split(string(output), "\n")
+	return splittedLine[0], splittedLine[1]
 }
 
 func main() {
+	// Create Integration
 	i, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
 	panicOnErr(err)
 
-	e := i.LocalEntity()
+	// Create Entity, entities name must be unique
+	entity := i.LocalEntity()
```

Comment:
Is the message above correct? I think there is only one local entity and it has no name. Maybe I'm wrong though